management/email_template_test.go

```diff
@@ -52,44 +53,62 @@ func TestEmailTemplateManager_Update(t *testing.T) {
 	template := givenAnEmailTemplate(t)
 
 	expectedBody := "<html><body><h1>Let's get you verified!</h1></body></html>"
+	expectedIncludeEmailInRedirect := false
 	err := m.EmailTemplate.Update(
 		template.GetTemplate(),
 		&EmailTemplate{
-			Body: &expectedBody,
+			Body:                   &expectedBody,
+			IncludeEmailInRedirect: &expectedIncludeEmailInRedirect,
 		},
 	)
 	assert.NoError(t, err)
 
 	actualTemplate, err := m.EmailTemplate.Read(template.GetTemplate())
 	assert.NoError(t, err)
 	assert.Equal(t, expectedBody, actualTemplate.GetBody())
+	assert.Equal(t, expectedIncludeEmailInRedirect, actualTemplate.GetIncludeEmailInRedirect())
 }
 
 func TestEmailTemplateManager_Replace(t *testing.T) {
 	setupHTTPRecordings(t)
 
+	err := m.Email.Delete()
+	assert.NoError(t, err)
```

Comment:
Deleting the email provider is necessary for repeatability. Otherwise, this will fail with a 409 when attempting to instantiate an email provider in `givenAnEmailProvider(t)` when one already exists. 

I chose to run in this test only to limit the blast radius of the change, but it may also reside in the `givenAnEmailProvider` helper function to help ensure the repeatability of other tests too.

Reply:
```suggestion
```

The email provider already gets deleted after the test: line 78 calls "`givenAnEmailProvider`" which calls the delete after the test: https://github.com/auth0/go-auth0/blob/bcd755ad5b80e1bc83fa3921764c30ce89a50928/management/email_test.go#L109-L111

Reply:
That works before the test, but if an email provider already exists, it also needs to occur _before_ a test is run. Otherwise it will perpetually fail.

Reply:
Ah, I think I get your point now. Let's do it differently tho, let's modify the `givenAnEmailProvider()` func to not error if there's already an email provider, but use that instead.